From 7ad208c83515c34f54200614e3ac6940a9dfb0fa Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <matthias.schiffer@tq-group.com>
Date: Mon, 2 Sep 2024 12:04:15 +0200
Subject: [PATCH] imx8m-hantro: implement
 imx_vpu_api_enc_get_skipped_frame_info() stubs for all encoders

Implement this function for all encoders to fix a linker error in
gstreamer-imx.

These encoders never return IMX_VPU_API_ENC_OUTPUT_CODE_FRAME_SKIPPED,
so we always return IMX_VPU_API_ENC_RETURN_CODE_INVALID_CALL from
imx_vpu_api_enc_get_skipped_frame_info().

Signed-off-by: Matthias Schiffer <matthias.schiffer@tq-group.com>
Upstream-Status: Submitted [https://github.com/Freescale/libimxvpuapi/pull/62]
---
 imxvpuapi2/imxvpuapi2_imx8m_hantro_dummy_encoder.c  | 10 ++++++++++
 imxvpuapi2/imxvpuapi2_imx8m_hantro_vc8000_encoder.c | 12 ++++++++++++
 2 files changed, 22 insertions(+)

diff --git a/imxvpuapi2/imxvpuapi2_imx8m_hantro_dummy_encoder.c b/imxvpuapi2/imxvpuapi2_imx8m_hantro_dummy_encoder.c
index 0e4d5e6..1568add 100644
--- a/imxvpuapi2/imxvpuapi2_imx8m_hantro_dummy_encoder.c
+++ b/imxvpuapi2/imxvpuapi2_imx8m_hantro_dummy_encoder.c
@@ -126,3 +126,13 @@ ImxVpuApiEncReturnCodes imx_vpu_api_enc_get_encoded_frame(ImxVpuApiEncoder *enco
 	IMX_VPU_API_UNUSED_PARAM(encoded_frame);
 	return IMX_VPU_API_ENC_RETURN_CODE_OK;
 }
+
+ImxVpuApiEncReturnCodes imx_vpu_api_enc_get_skipped_frame_info(ImxVpuApiEncoder *encoder, void **context, uint64_t *pts, uint64_t *dts)
+{
+	IMX_VPU_API_UNUSED_PARAM(encoder);
+	IMX_VPU_API_UNUSED_PARAM(context);
+	IMX_VPU_API_UNUSED_PARAM(pts);
+	IMX_VPU_API_UNUSED_PARAM(dts);
+
+	return IMX_VPU_API_ENC_RETURN_CODE_INVALID_CALL;
+}
diff --git a/imxvpuapi2/imxvpuapi2_imx8m_hantro_vc8000_encoder.c b/imxvpuapi2/imxvpuapi2_imx8m_hantro_vc8000_encoder.c
index 42f3a8f..09f2faf 100644
--- a/imxvpuapi2/imxvpuapi2_imx8m_hantro_vc8000_encoder.c
+++ b/imxvpuapi2/imxvpuapi2_imx8m_hantro_vc8000_encoder.c
@@ -1426,3 +1426,15 @@ ImxVpuApiEncReturnCodes imx_vpu_api_enc_get_encoded_frame_ext(ImxVpuApiEncoder *
 
 	return IMX_VPU_API_ENC_RETURN_CODE_OK;
 }
+
+ImxVpuApiEncReturnCodes imx_vpu_api_enc_get_skipped_frame_info(ImxVpuApiEncoder *encoder, void **context, uint64_t *pts, uint64_t *dts)
+{
+	IMX_VPU_API_UNUSED_PARAM(encoder);
+	IMX_VPU_API_UNUSED_PARAM(context);
+	IMX_VPU_API_UNUSED_PARAM(pts);
+	IMX_VPU_API_UNUSED_PARAM(dts);
+
+	/* Frameskipping with VC8000 is not supported */
+
+	return IMX_VPU_API_ENC_RETURN_CODE_INVALID_CALL;
+}
